### ISSUE TITLE
Similar project for state legislatures?

### ISSUE THREAD

**Issue by halffullheart** [jlgoldman/membersofcongress#4]
This project is fantastic. Thanks so much for building this. I’m using it in one of my own projects and I find that I’ll probably want similar data—district offices included—for state legislatures. I live in WA state, and would probably be most interested in starting there. Taking on something like this is obviously much bigger than doing just US reps. Are you interested in working together on something like that?

**Comment by jlgoldman** [comment]
Thanks! Yeah, I'd love to see this grow and have a community of supporters/maintainers.

The best source for federal-level politicians is https://github.com/unitedstates/congress-legislators, and it was only because that repo didn't have district office data that I created this one. At the same time I created this data set, a fork of that one was created in https://github.com/TheWalkers/congress-legislators which also contains district office data. I used my data to help clean, validate, and complete that data set, so both should be comparable.

Ideally, everything would wind up in https://github.com/unitedstates/congress-legislators, but as the maintainer of that is overburdened it's unlikely that that will grow to state-level data. It would be worth gauging whether the maintainers of https://github.com/TheWalkers/congress-legislators are interested in state-level, and if so that repo could be a good choice because it's in the same format as the unitedstates one which may be what people are used to using.

But if you prefer this repo or they're not interested in state-level data, I'm happy to see this grow to contain that data. I just think that state-level data should be in a separate file and tagged WIP so consumers can see at a glance which data is essentially complete and which is still being gathered.

How were you looking to get data, both for listing the reps themselves as well as their contact info? I built a clipping tool for this repo to make it fast to manually pull contact info from websites, I could try to release that to be reusable.

**Comment by halffullheart** [comment]
Thanks for the background. I had already poked around on those two repos you mentioned but I wasn’t sure if the data from this repo was going to make it’s way in permanently (or at all, in the case of the unitedstates one). Also reading about your process, I felt like I could trust the data here more, but I agree it’s probably comparable. Usually in cases like this it’s important to just go with one and try to rally people around making it excellent.

As far as the state specific data goes, I agree separate files for each state would make sense. I was imagining working in a fork or branch until there was at least data for every legislator’s main office. I think/hope for a lot of states there is public data that can be reformatted, and it will probably just have the basics—one address and phone number in the capitol. At that point it would be safe, I think, to merge into the main branch and either do that process again for adding district office data or wait until it’s somewhat complete before merging.

Your clipping tool sounds pretty great. I have dreams of crowdsourcing data collection by providing an editable Google spreadsheet (so non-technical folks would feel they can participate) and then dumping it out to other formats later. But who knows if there are really any people who would want to do this kind of boring work.

I’ll reach out to the folks at TheWalkers and see if they have any interest in state level data, and do more research on availability of public data for the states.

**Comment by jlgoldman** [comment]
Awesome, thanks for the info. If you go wit this repo I'm actually fine having the partial data in the main branch as long as it's labeled as partial. I'm all for actionable data soon rather than perfect data later.

Let me know as you come across data sources and I'll see if the clipper can be made to work with them. Ideally I'd get it to the point where somebody non-technical could use it too, though that may be a few weeks away.

**Comment by halffullheart** [comment]
I did some research and I have good news: Open States ([website](https://openstates.org/), [github](https://github.com/openstates/)) already has state legislature data for all 50 states, Puerto Rico, and D.C.

I also discovered that district offices are not as common for state legislators as they are for members of congress. For example, in WA, only some legislators have a district office, and many of them are shared between several legislators.

Open States does have district offices in some cases. I’m not sure how good the coverage is but it looks like a lot of work has gone into the project.

**Comment by jlgoldman** [comment]
Oh great, good find! So I take it you think the data is largely usable as-is, and where it's not we should contribute to that repo?

**Comment by halffullheart** [comment]
Yes, I think that’s a good plan. Open State works mostly using scrapers, so it’s not as simple as just manually adding any missing data.

For the app I’m working on, I plan to use [Google’s Civic Information API](https://developers.google.com/civic-information/) to look up legislators since that API is the only one I have found that:

1. Allows lookups by lat/long or address
2. Returns legislators at the state and US level

Google does not seem to have high quality district office data, so separately I will store my own district office data taken from either this repo or TheWalkers/congress-legislators for US level and Open Sates for state level and use that to augment the results from Google.

Thanks for entertaining my crazy ideas here. Are you working on any projects that use this data? I’m working on a tool that would let people create contact sheets with a script or talking points paired with the user’s specific representative contact info. For example the creator would specify that they want WA State Senators to be displayed. Someone using their sheet would see *their senator* (after they put in their address, or use their device’s location).

**Comment by jlgoldman** [comment]
Awesome, sounds like a great project.

Yeah it's frustrating that there's no single API that has all the information needed to build many types of apps. But it's hard info to collect and maintain and there's no money in it, so I guess that's just how it is.

I have a project on the backburner that makes use of (federal) district office data. I noticed that lots of sites were building their forms for finding one's representatives and their contact info searching by address, so I wanted to create a suite of embeddable widgets that people could just plug into their sites instead of building their own. The first example of it is sort-of live but not really finished.

http://civic-widgets.herokuapp.com/gallery

I also wanted to incorporate district office data into https://www.writetogov.com/ but haven't gotten around to it yet. Part of it was that I learned when collecting district office data that most mail sent to a district office is forwarded to the DC office, and screened for security, and could take a long time to arrive. Since the site is centered around writing, with that unfortunate info it didn't necessarily seem like much of a gain to have district addresses. Oh well.
